src/webgpu/api/operation/adapter/requestDevice.spec.ts

```diff
@@ -1,16 +1,106 @@
 import { GPUTest } from '../../../gpu_test.js';
 
 export const g = makeTestGroup(GPUTest);
 
+const powerPreferenceModes: Array<GPUPowerPreference | undefined> = [
+  undefined,
+  'low-power',
+  'high-performance',
+];
+const forceFallbackOptions: Array<boolean | undefined> = [undefined, false, true];
+
+async function testAdapter(adapter: GPUAdapter | null) {
+  assert(adapter !== null, 'Failed to get adapter.');
+  const device = await adapter.requestDevice();
+  assert(device !== null, 'Failed to get device.');
+  const kOffset = 1230000;
+  const pipeline = await device.createComputePipeline({
+    compute: {
+      module: device.createShaderModule({
+        code: `
+          struct Buffer { data: array<u32>; };
+
+          @group(0) @binding(0) var<storage, read_write> buffer: Buffer;
+          @stage(compute) @workgroup_size(1u) fn main(
+              @builtin(global_invocation_id) id: vec3<u32>) {
+            buffer.data[id.x] = id.x + ${kOffset}u;
+          }
+        `,
+      }),
+      entryPoint: 'main',
+    },
+  });
+  const kNumElements = 64;
+  const kBufferSize = kNumElements * 4;
+  const buffer = device.createBuffer({
+    size: kBufferSize,
+    usage: GPUBufferUsage.STORAGE | GPUBufferUsage.COPY_SRC,
+  });
+  const resultBuffer = device.createBuffer({
+    size: kBufferSize,
+    usage: GPUBufferUsage.MAP_READ | GPUBufferUsage.COPY_DST,
+  });
+  const bindGroup = device.createBindGroup({
+    layout: pipeline.getBindGroupLayout(0),
+    entries: [{ binding: 0, resource: { buffer } }],
+  });
+  const encoder = device.createCommandEncoder();
+  const pass = encoder.beginComputePass();
+  pass.setPipeline(pipeline);
+  pass.setBindGroup(0, bindGroup);
+  pass.dispatch(kNumElements);
+  pass.end();
+  encoder.copyBufferToBuffer(buffer, 0, resultBuffer, 0, kBufferSize);
+  device.queue.submit([encoder.finish()]);
+  const expected = new Uint32Array([...iterRange(kNumElements, x => x + kOffset)]);
+  await resultBuffer.mapAsync(GPUMapMode.READ);
+  const actual = new Uint32Array(resultBuffer.getMappedRange());
+  assert(objectEquals(actual, expected), 'compute pipeline ran');
+  resultBuffer.unmap();
+  resultBuffer.destroy();
+  buffer.destroy();
+  device.destroy();
+}
+
+g.test('requestAdaptor')
+  .desc(`request adapter with all possible options and check for basic functionality`)
+  .params(u =>
+    u
+      .combine('powerPreference', powerPreferenceModes)
+      .combine('forceFallbackAdapter', forceFallbackOptions)
+  )
+  .fn(async t => {
+    const { powerPreference, forceFallbackAdapter } = t.params;
+    const adapter = await getGPU().requestAdapter({
+      ...(powerPreference !== undefined && { powerPreference }),
+      ...(forceFallbackAdapter !== undefined && { forceFallbackAdapter }),
+    });
+    // failing to create an adapter when forceFallbackAdapter is true is ok.
+    if (forceFallbackAdapter && !adapter) {
+      return;
```

Comment:
```
t.skip('No adapter available');
return;
``` 

might be more expressive. This is what we do for tests that lack extension support.